test/sharness/lib/test-lib.sh

```diff
@@ -53,7 +94,7 @@ TEST_OS="$(uname -s | tr '[a-z]' '[A-Z]')"
 test "$TEST_NO_FUSE" != 1 && test_set_prereq FUSE
 test "$TEST_EXPENSIVE" = 1 && test_set_prereq EXPENSIVE
 test "$TEST_NO_DOCKER" != 1 && type docker >/dev/null 2>&1 && groups | egrep "\bdocker\b" && test_set_prereq DOCKER
-test "$TEST_NO_PLUGIN" != 1 && test "$TEST_OS" = "LINUX" && test_set_prereq PLUGIN
+test "$TEST_NO_PLUGIN" != 1 && test_have_prereq FUSE && test "$TEST_OS" = "LINUX" && test_set_prereq PLUGIN
```

Comment:
FUSE is not a prereq for the plugin tests.

Reply:
I am sorry I should have used **way** more words in the PR description. This is similar ( but not the same as ) https://github.com/ipfs/go-ds-s3/issues/63. You are not running into it on CI because you test with both NO_FUSE and NO_PLUGIN at the same time. Observe below the `FAILING` case ( if I **remove** `TEST_NO_FUSE=1` and clean up everything woks as expected ):
```
ipfs:~/ramdisk/go-ipfs$ git rev-parse HEAD
5185e8e859519099782ccff12f7ceec057b760ef
```
```
ipfs:~/ramdisk/go-ipfs/test/sharness$ git diff -U1
diff --git a/test/sharness/t0280-plugin.sh b/test/sharness/t0280-plugin.sh
index a0709ef1b..a499119d5 100755
--- a/test/sharness/t0280-plugin.sh
+++ b/test/sharness/t0280-plugin.sh
@@ -36,2 +36,10 @@ test_expect_success "cleanup bad plugin" '
 
+echo  go build \
+    -asmflags=all="-trimpath=${GOPATH}" -gcflags=all="-trimpath=${GOPATH}" \
+    -buildmode=plugin -o "$IPFS_PATH/plugins/example.so" ../t0280-plugin-data/example.go
+
+  go build \
+    -asmflags=all="-trimpath=${GOPATH}" -gcflags=all="-trimpath=${GOPATH}" \
+    -buildmode=plugin -o "$IPFS_PATH/plugins/example.so" ../t0280-plugin-data/example.go
+
 test_expect_success "install test plugin" '
@@ -43,2 +51,4 @@ test_expect_success "install test plugin" '
 
+ipfs id
+
 test_plugin() {
```
```
ipfs:~/ramdisk/go-ipfs$ export TEST_NO_FUSE=1
```
```
ipfs:~/ramdisk/go-ipfs$ chmod u+w -R ~/go test/sharness/trash\ directory.t0280-plugin.sh/
ipfs:~/ramdisk/go-ipfs$ rm -rf ~/go test/sharness/trash\ directory.t0280-plugin.sh/
```
```
ipfs:~/ramdisk/go-ipfs$ make build
go: downloading github.com/ipfs/go-ipfs-cmds v0.1.1
go: downloading github.com/libp2p/go-socket-activation v0.0.2
go: downloading github.com/ipfs/go-log v1.0.2
go: downloading github.com/jbenet/goprocess v0.1.3
go: downloading github.com/ipfs/go-fs-lock v0.0.4
go: downloading github.com/multiformats/go-multiaddr-net v0.1.3
go: downloading github.com/ipfs/go-datastore v0.4.4
go: downloading github.com/ipfs/go-ipfs-files v0.0.6
go: downloading github.com/ipfs/go-metrics-prometheus v0.0.2
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/ipfs/go-path v0.0.7
go: downloading github.com/coreos/go-systemd v0.0.0-20181012123002-c6f51f82210d
go: downloading github.com/ipfs/go-mfs v0.1.1
go: downloading go4.org v0.0.0-20200104003542-c7e774b10ea0
go: downloading github.com/whyrusleeping/base32 v0.0.0-20170828182744-c30ac30633cc
go: downloading github.com/libp2p/go-libp2p-http v0.1.5
go: downloading github.com/prometheus/client_golang v1.5.0
go: downloading github.com/multiformats/go-multibase v0.0.1
go: downloading github.com/ipfs/go-ipns v0.0.2
go: downloading github.com/whyrusleeping/tar-utils v0.0.0-20180509141711-8c6c8ba81d5c
go: downloading github.com/mr-tron/base58 v1.1.3
go: downloading github.com/libp2p/go-libp2p-loggables v0.1.0
go: downloading github.com/coreos/go-systemd/v22 v22.0.0
go: downloading github.com/ipfs/go-graphsync v0.0.5
go: downloading github.com/opentracing/opentracing-go v1.1.0
go: downloading github.com/ipfs/go-metrics-interface v0.0.1
go: downloading github.com/ipfs/go-ipfs-pinner v0.0.4
go: downloading github.com/ipfs/go-ipfs-exchange-offline v0.0.1
go: downloading github.com/libp2p/go-libp2p-kad-dht v0.5.1
go: downloading github.com/ipfs/go-cid v0.0.5
go: downloading github.com/libp2p/go-libp2p-core v0.5.0
go: downloading github.com/ipfs/go-unixfs v0.2.4
go: downloading github.com/ipfs/go-filestore v0.0.3
go: downloading github.com/ipfs/go-ipfs-util v0.0.1
go: downloading github.com/ipfs/go-log/v2 v2.0.2
go: downloading gopkg.in/cheggaaa/pb.v1 v1.0.28
go: downloading github.com/multiformats/go-base32 v0.0.3
go: downloading go.uber.org/fx v1.10.0
go: downloading github.com/ipfs/interface-go-ipfs-core v0.2.6
go: downloading github.com/libp2p/go-libp2p-record v0.1.2
go: downloading github.com/libp2p/go-msgio v0.0.4
go: downloading github.com/multiformats/go-multiaddr v0.2.1
go: downloading github.com/libp2p/go-libp2p-peerstore v0.2.0
go: downloading github.com/elgris/jsondiff v0.0.0-20160530203242-765b5c24c302
go: downloading github.com/ipfs/go-bitswap v0.2.2
go: downloading github.com/ipfs/go-ipfs-config v0.2.1
go: downloading github.com/jbenet/go-temp-err-catcher v0.1.0
go: downloading github.com/libp2p/go-libp2p v0.6.0
go: downloading github.com/libp2p/go-libp2p-kbucket v0.2.3
go: downloading github.com/multiformats/go-multiaddr-dns v0.2.0
go: downloading github.com/ipfs/go-ipfs-ds-help v0.1.1
go: downloading github.com/google/uuid v1.1.1
go: downloading github.com/whyrusleeping/go-sysinfo v0.0.0-20190219211824-4a357d4b90b1
go: downloading github.com/bren2010/proquint v0.0.0-20160323162903-38337c27106d
go: downloading github.com/ipfs/go-todocounter v0.0.2
go: downloading github.com/ipfs/go-ipfs-routing v0.1.0
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/btcsuite/btcd v0.20.1-beta
go: downloading github.com/libp2p/go-libp2p-connmgr v0.2.1
go: downloading github.com/ipfs/go-ds-measure v0.1.0
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/cskr/pubsub v1.0.2
go: downloading github.com/libp2p/go-libp2p-quic-transport v0.3.1
go: downloading github.com/libp2p/go-buffer-pool v0.0.2
go: downloading github.com/libp2p/go-libp2p-routing v0.1.0
go: downloading github.com/ipfs/go-ipfs-exchange-interface v0.0.1
go: downloading github.com/libp2p/go-libp2p-pubsub v0.2.6
go: downloading github.com/mattn/go-runewidth v0.0.8
go: downloading github.com/multiformats/go-varint v0.0.5
go: downloading github.com/ipfs/go-block-format v0.0.2
go: downloading github.com/whyrusleeping/go-keyspace v0.0.0-20160322163242-5b898ac5add1
go: downloading github.com/ipfs/go-blockservice v0.1.2
go: downloading go.opencensus.io v0.22.3
go: downloading github.com/libp2p/go-libp2p-circuit v0.1.4
go: downloading github.com/ipfs/go-cidutil v0.0.2
go: downloading github.com/ipfs/go-ipfs-delay v0.0.1
go: downloading github.com/multiformats/go-multistream v0.1.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/Kubuxu/go-os-helper v0.0.1
go: downloading golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/ipfs/go-ipfs-posinfo v0.0.1
go: downloading golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
go: downloading github.com/ipfs/go-merkledag v0.3.1
go: downloading go.uber.org/zap v1.10.0
go: downloading github.com/ipfs/go-ipfs-chunker v0.0.4
go: downloading github.com/vishvananda/netlink v1.1.0
go: downloading github.com/libp2p/go-libp2p-peer v0.2.0
go: downloading github.com/Stebalien/go-bitfield v0.0.1
go: downloading github.com/ipfs/go-peertaskqueue v0.2.0
go: downloading go.uber.org/dig v1.8.0
go: downloading github.com/ipfs/go-ipld-format v0.0.2
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/libp2p/go-libp2p-secio v0.2.1
go: downloading golang.org/x/crypto v0.0.0-20200311171314-f7b00557c8c4
go: downloading github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
go: downloading github.com/ipfs/go-ipfs-provider v0.4.1
go: downloading github.com/syndtr/goleveldb v1.0.0
go: downloading github.com/libp2p/go-libp2p-crypto v0.1.0
go: downloading github.com/ipfs/go-verifcid v0.0.1
go: downloading go.uber.org/atomic v1.5.0
go: downloading github.com/libp2p/go-libp2p-autonat-svc v0.1.0
go: downloading github.com/prometheus/common v0.9.1
go: downloading github.com/multiformats/go-multiaddr-fmt v0.1.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/libp2p/go-libp2p-tls v0.1.3
go: downloading github.com/vishvananda/netns v0.0.0-20191106174202-0a2b9b5464df
go: downloading github.com/mattn/go-colorable v0.1.4
go: downloading github.com/libp2p/go-libp2p-transport-upgrader v0.2.0
go: downloading github.com/libp2p/go-libp2p-yamux v0.2.3
go: downloading github.com/whyrusleeping/timecache v0.0.0-20160911033111-cfcb2f1abfee
go: downloading github.com/libp2p/go-libp2p-autonat v0.1.1
go: downloading github.com/ipfs/go-ipfs-pq v0.0.2
go: downloading github.com/ipfs/go-ipfs-blockstore v0.1.4
go: downloading github.com/texttheater/golang-levenshtein v0.0.0-20180516184445-d188e65d659e
go: downloading github.com/multiformats/go-multihash v0.0.13
go: downloading github.com/ipfs/go-ds-leveldb v0.4.2
go: downloading github.com/ipld/go-ipld-prime v0.0.2-0.20191108012745-28a82f04c785
go: downloading github.com/libp2p/go-libp2p-discovery v0.2.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/libp2p/go-libp2p-routing-helpers v0.1.0
go: downloading github.com/libp2p/go-ws-transport v0.2.0
go: downloading github.com/libp2p/go-yamux v1.3.1
go: downloading github.com/ipfs/go-ds-flatfs v0.4.0
go: downloading github.com/libp2p/go-flow-metrics v0.0.3
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/ipfs/go-ipld-git v0.0.3
go: downloading github.com/whyrusleeping/multiaddr-filter v0.0.0-20160516205228-e903e4adabd7
go: downloading github.com/ipfs/go-ipld-cbor v0.0.4
go: downloading github.com/mattn/go-isatty v0.0.11
go: downloading github.com/libp2p/go-tcp-transport v0.1.1
go: downloading github.com/whyrusleeping/mdns v0.0.0-20190826153040-b9b60ed33aa9
go: downloading github.com/ipfs/bbloom v0.0.4
go: downloading github.com/libp2p/go-libp2p-pubsub-router v0.2.1
go: downloading github.com/facebookgo/atomicfile v0.0.0-20151019160806-2de1f203e7d5
go: downloading github.com/libp2p/go-eventbus v0.1.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/minio/blake2b-simd v0.0.0-20160723061019-3f5f724cb5b1
go: downloading github.com/whyrusleeping/chunker v0.0.0-20181014151217-fe64bd25879f
go: downloading github.com/lucas-clemente/quic-go v0.15.2
go: downloading github.com/libp2p/go-maddr-filter v0.0.5
go: downloading github.com/gorilla/websocket v1.4.1
go: downloading github.com/libp2p/go-libp2p-pnet v0.2.0
go: downloading github.com/polydawn/refmt v0.0.0-20190408063855-01bf1e26dd14
go: downloading github.com/libp2p/go-libp2p-nat v0.0.5
go: downloading github.com/libp2p/go-reuseport v0.0.1
go: downloading golang.org/x/net v0.0.0-20190724013045-ca1201d0de80
go: downloading github.com/libp2p/go-libp2p-swarm v0.2.2
go: downloading go.uber.org/multierr v1.4.0
go: downloading github.com/whyrusleeping/cbor-gen v0.0.0-20200123233031-1cdf64d27158
go: downloading github.com/jbenet/go-is-domain v1.0.3
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/prometheus/procfs v0.0.8
go: downloading github.com/libp2p/go-reuseport-transport v0.0.2
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/davidlazar/go-crypto v0.0.0-20170701192655-dcfb0a7ac018
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/libp2p/go-stream-muxer-multistream v0.2.0
go: downloading github.com/ipfs/go-ds-badger v0.2.1
go: downloading github.com/francoispqt/gojay v1.2.13
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/libp2p/go-nat v0.0.4
go: downloading github.com/libp2p/go-libp2p-mplex v0.2.2
go: downloading github.com/cheekybits/genny v1.0.0
go: downloading github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: downloading github.com/libp2p/go-libp2p-gostream v0.2.1
go: downloading github.com/marten-seemann/qtls v0.8.0
go: downloading github.com/libp2p/go-conn-security-multistream v0.1.0
go: downloading github.com/libp2p/go-addr-util v0.0.1
go: downloading github.com/miekg/dns v1.1.12
go: downloading github.com/dgraph-io/badger v1.6.0
go: downloading github.com/huin/goupnp v1.0.0
go: downloading github.com/jackpal/gateway v1.0.5
go: downloading github.com/libp2p/go-mplex v0.1.1
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/koron/go-ssdp v0.0.0-20191105050749-2e1c40ed0b5d
go: downloading github.com/jackpal/go-nat-pmp v1.0.1
go: downloading github.com/ipld/go-ipld-prime-proto v0.0.0-20191113031812-e32bd156a1e5
go: downloading golang.org/x/text v0.3.2
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/whyrusleeping/mafmt v1.2.8
go: downloading github.com/dgryski/go-farm v0.0.0-20190423205320-6a90982ecee2
go: downloading github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96
plugin/loader/preload.sh > plugin/loader/preload.go
go version go1.14 linux/amd64
bin/check_go_version 1.14
go fmt plugin/loader/preload.go >/dev/null
go: downloading bazil.org/fuse v0.0.0-20200117225306-7b5117fecadc
go build  "-asmflags=all='-trimpath='" "-gcflags=all='-trimpath='" -ldflags="-X "github.com/ipfs/go-ipfs".CurrentCommit=5185e8e85-dirty" -tags="nofuse" -o "cmd/ipfs/ipfs" "github.com/ipfs/go-ipfs/cmd/ipfs"
```
```
ipfs:~/ramdisk/go-ipfs$ cd test/sharness/
ipfs:~/ramdisk/go-ipfs/test/sharness$ bash t0280-plugin.sh
ok 1 - ipfs init succeeds
ok 2 - prepare config -- mounting
ok 3 - ipfs id succeeds
ok 4 - make a bad plugin
ok 5 - ipfs id fails due to a bad plugin
ok 6 - cleanup bad plugin
go build -asmflags=all=-trimpath= -gcflags=all=-trimpath= -buildmode=plugin -o /home/ipfs/ramdisk/go-ipfs/test/sharness/trash directory.t0280-plugin.sh/.ipfs/plugins/example.so ../t0280-plugin-data/example.go
go: downloading github.com/opentracing/opentracing-go v1.1.0
go: downloading github.com/ipfs/go-ipld-format v0.0.2
go: downloading github.com/ipfs/interface-go-ipfs-core v0.2.6
go: downloading github.com/ipfs/go-datastore v0.4.4
go: downloading github.com/ipfs/go-blockservice v0.1.2
go: downloading github.com/ipfs/go-ipfs-config v0.2.1
go: downloading github.com/ipfs/go-ds-measure v0.1.0
go: downloading github.com/ipfs/go-fs-lock v0.0.4
go: downloading github.com/ipfs/go-ipfs-provider v0.4.1
go: downloading github.com/ipfs/go-ipfs-pinner v0.0.4
go: downloading github.com/ipfs/go-cid v0.0.5
go: downloading github.com/libp2p/go-libp2p-core v0.5.0
go: downloading github.com/libp2p/go-libp2p-autonat-svc v0.1.0
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/jbenet/goprocess v0.1.3
go: downloading github.com/libp2p/go-libp2p-pubsub v0.2.6
go: downloading github.com/multiformats/go-multiaddr-dns v0.2.0
go: downloading github.com/multiformats/go-varint v0.0.5
go: downloading github.com/libp2p/go-libp2p-autonat v0.1.1
go: downloading github.com/ipfs/go-ipfs-routing v0.1.0
go: downloading github.com/multiformats/go-multiaddr-net v0.1.3
go: downloading github.com/ipfs/go-ipfs-files v0.0.6
go: downloading github.com/ipfs/go-merkledag v0.3.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/ipfs/go-path v0.0.7
go: downloading github.com/ipfs/go-graphsync v0.0.5
go: downloading github.com/ipfs/go-log v1.0.2
go: downloading golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
go: downloading go4.org v0.0.0-20200104003542-c7e774b10ea0
go: downloading github.com/jbenet/go-is-domain v1.0.3
go: downloading github.com/multiformats/go-multistream v0.1.1
go: downloading github.com/google/uuid v1.1.1
go: downloading github.com/multiformats/go-multihash v0.0.13
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/ipfs/go-bitswap v0.2.2
go: downloading github.com/libp2p/go-libp2p-discovery v0.2.0
go: downloading github.com/libp2p/go-libp2p v0.6.0
go: downloading github.com/ipfs/go-verifcid v0.0.1
go: downloading github.com/bren2010/proquint v0.0.0-20160323162903-38337c27106d
go: downloading github.com/libp2p/go-libp2p-record v0.1.2
go: downloading github.com/ipfs/go-filestore v0.0.3
go: downloading go.uber.org/fx v1.10.0
go: downloading github.com/ipfs/go-mfs v0.1.1
go: downloading go.uber.org/zap v1.10.0
go: downloading github.com/libp2p/go-libp2p-kad-dht v0.5.1
go: downloading github.com/ipfs/go-ipfs-ds-help v0.1.1
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/ipfs/go-ipns v0.0.2
go: downloading github.com/ipfs/go-peertaskqueue v0.2.0
go: downloading github.com/ipfs/go-block-format v0.0.2
go: downloading github.com/ipfs/go-metrics-interface v0.0.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/ipfs/go-ipfs-util v0.0.1
go: downloading golang.org/x/crypto v0.0.0-20200311171314-f7b00557c8c4
go: downloading go.uber.org/multierr v1.4.0
go: downloading github.com/libp2p/go-libp2p-peerstore v0.2.0
go: downloading github.com/ipld/go-ipld-prime v0.0.2-0.20191108012745-28a82f04c785
go: downloading go.uber.org/atomic v1.5.0
go: downloading github.com/libp2p/go-libp2p-loggables v0.1.0
go: downloading github.com/ipfs/go-log/v2 v2.0.2
go: downloading github.com/multiformats/go-multibase v0.0.1
go: downloading github.com/libp2p/go-msgio v0.0.4
go: downloading github.com/ipfs/go-unixfs v0.2.4
go: downloading github.com/ipfs/go-todocounter v0.0.2
go: downloading github.com/mr-tron/base58 v1.1.3
go: downloading go.opencensus.io v0.22.3
go: downloading github.com/ipfs/go-ipfs-exchange-offline v0.0.1
go: downloading github.com/whyrusleeping/timecache v0.0.0-20160911033111-cfcb2f1abfee
go: downloading github.com/minio/blake2b-simd v0.0.0-20160723061019-3f5f724cb5b1
go: downloading github.com/ipfs/go-ipfs-chunker v0.0.4
go: downloading github.com/cskr/pubsub v1.0.2
go: downloading bazil.org/fuse v0.0.0-20200117225306-7b5117fecadc
go: downloading github.com/ipfs/go-ipfs-delay v0.0.1
go: downloading github.com/libp2p/go-buffer-pool v0.0.2
go: downloading golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
go: downloading github.com/facebookgo/atomicfile v0.0.0-20151019160806-2de1f203e7d5
go: downloading github.com/libp2p/go-flow-metrics v0.0.3
go: downloading github.com/ipfs/go-ipld-cbor v0.0.4
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/libp2p/go-libp2p-nat v0.0.5
go: downloading go.uber.org/dig v1.8.0
go: downloading github.com/multiformats/go-multiaddr v0.2.1
go: downloading github.com/libp2p/go-maddr-filter v0.0.5
go: downloading github.com/libp2p/go-libp2p-secio v0.2.1
go: downloading github.com/libp2p/go-libp2p-peer v0.2.0
go: downloading github.com/libp2p/go-libp2p-pubsub-router v0.2.1
go: downloading github.com/multiformats/go-base32 v0.0.3
go: downloading github.com/ipfs/go-ipfs-posinfo v0.0.1
go: downloading github.com/ipfs/go-ipfs-blockstore v0.1.4
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/libp2p/go-libp2p-circuit v0.1.4
go: downloading github.com/libp2p/go-ws-transport v0.2.0
go: downloading github.com/whyrusleeping/mdns v0.0.0-20190826153040-b9b60ed33aa9
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/polydawn/refmt v0.0.0-20190408063855-01bf1e26dd14
go: downloading github.com/libp2p/go-libp2p-transport-upgrader v0.2.0
go: downloading github.com/whyrusleeping/cbor-gen v0.0.0-20200123233031-1cdf64d27158
go: downloading github.com/libp2p/go-nat v0.0.4
go: downloading github.com/libp2p/go-libp2p-kbucket v0.2.3
go: downloading github.com/whyrusleeping/chunker v0.0.0-20181014151217-fe64bd25879f
go: downloading github.com/jbenet/go-temp-err-catcher v0.1.0
go: downloading github.com/libp2p/go-eventbus v0.1.0
go: downloading github.com/libp2p/go-conn-security-multistream v0.1.0
go: downloading github.com/libp2p/go-libp2p-swarm v0.2.2
go: downloading github.com/ipfs/go-ipfs-exchange-interface v0.0.1
go: downloading github.com/multiformats/go-multiaddr-fmt v0.1.0
go: downloading github.com/libp2p/go-libp2p-yamux v0.2.3
go: downloading github.com/gorilla/websocket v1.4.1
go: downloading github.com/libp2p/go-libp2p-crypto v0.1.0
go: downloading github.com/ipfs/bbloom v0.0.4
go: downloading github.com/jackpal/go-nat-pmp v1.0.1
go: downloading github.com/ipfs/go-ipfs-pq v0.0.2
go: downloading github.com/whyrusleeping/go-keyspace v0.0.0-20160322163242-5b898ac5add1
go: downloading github.com/huin/goupnp v1.0.0
go: downloading github.com/libp2p/go-libp2p-mplex v0.2.2
go: downloading github.com/libp2p/go-libp2p-pnet v0.2.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/libp2p/go-addr-util v0.0.1
go: downloading github.com/ipfs/go-cidutil v0.0.2
go: downloading github.com/jackpal/gateway v1.0.5
go: downloading github.com/libp2p/go-libp2p-routing v0.1.0
go: downloading github.com/Stebalien/go-bitfield v0.0.1
go: downloading github.com/libp2p/go-tcp-transport v0.1.1
go: downloading github.com/davidlazar/go-crypto v0.0.0-20170701192655-dcfb0a7ac018
go: downloading golang.org/x/net v0.0.0-20190724013045-ca1201d0de80
go: downloading github.com/whyrusleeping/multiaddr-filter v0.0.0-20160516205228-e903e4adabd7
go: downloading github.com/ipld/go-ipld-prime-proto v0.0.0-20191113031812-e32bd156a1e5
go: downloading github.com/libp2p/go-libp2p-connmgr v0.2.1
go: downloading github.com/miekg/dns v1.1.12
go: downloading github.com/whyrusleeping/base32 v0.0.0-20170828182744-c30ac30633cc
go: downloading github.com/libp2p/go-libp2p-tls v0.1.3
go: downloading github.com/libp2p/go-reuseport v0.0.1
go: downloading github.com/btcsuite/btcd v0.20.1-beta
go: downloading github.com/libp2p/go-yamux v1.3.1
go: downloading github.com/libp2p/go-libp2p-routing-helpers v0.1.0
go: downloading github.com/libp2p/go-stream-muxer-multistream v0.2.0
go: downloading github.com/libp2p/go-reuseport-transport v0.0.2
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/koron/go-ssdp v0.0.0-20191105050749-2e1c40ed0b5d
go: downloading github.com/libp2p/go-mplex v0.1.1
go: downloading golang.org/x/text v0.3.2
go: downloading github.com/whyrusleeping/mafmt v1.2.8
go: downloading github.com/libp2p/go-libp2p-quic-transport v0.3.1
go: downloading github.com/lucas-clemente/quic-go v0.15.2
go: downloading github.com/vishvananda/netlink v1.1.0
go: downloading github.com/vishvananda/netns v0.0.0-20191106174202-0a2b9b5464df
go: downloading github.com/cheekybits/genny v1.0.0
go: downloading github.com/francoispqt/gojay v1.2.13
go: downloading github.com/marten-seemann/qtls v0.8.0
go: downloading github.com/golang/protobuf v1.3.2
ok 7 - install test plugin
Error: error loading plugins: loading plugin /home/ipfs/ramdisk/go-ipfs/test/sharness/trash directory.t0280-plugin.sh/.ipfs/plugins/example.so: plugin.Open("/home/ipfs/ramdisk/go-ipfs/test/sharness/trash directory.t0280-plugin.sh/.ipfs/plugins/example"): plugin was built with a different version of package github.com/multiformats/go-varint
not ok 8 - id runs
...
```

Reply:
Also attaching the "happy path" with the exact same setup while I have it:

`< time passes >`

Ok... W.T.F.... it now fails exactly the same way **without** FUSE disabled. I swear I did test this last night before pushing the PR...

HALP

Reply:
@Stebalien note - this is not a blocker, we can keep this open until after the crunch
I also have this on a standalone VM, I can just drop your key on there so you can login and see it fail yourself

Reply:
Changing an unrelated variable because it happens to fix the tests is never the right solution.

Reply:
> Changing an unrelated variable because it happens to fix

This is an unfair characterization of what happened. Originally I was able to pass the test with fuse enabled everywhere, thus the ( now incorrect ) thinking that `-tags="nofuse"` being present in one spot but not the other is the culprit.

Now I know this is not the case, but the point remains that I can't pass this test on a relatively standard box.

I am pulling out the offending commit from this PR so we can merge the rest, will circle back to this in a day.

Reply:
> Originally I was able to pass the test with fuse enabled everywhere, thus the ( now incorrect ) thinking that `-tags="nofuse"` being present in one spot but not the other is the culprit.

My concern is that we have no idea _why_ that mattered. We need to make sure to address the underlying problem instead of addressing the symptom.